helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -322,8 +493,24 @@ public Object removeRoutingTableChangeListener(
    */
   public List<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName,
       String partitionName, String state, List<String> resourceTags) {
-    return _routingTableRef.get().getInstancesForResourceGroup(resourceGroupName, partitionName,
-        state, resourceTags);
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getInstancesForResourceGroup(resourceGroupName,
+        partitionName, state, resourceTags);
+  }
+
+  public List<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName,
```

Comment:
Same here. If not in the design, let's hold this for now.

Reply:
Removed.